### DISCUSSION TITLE
Is it possible to read value that changes in less than 50ms interval?

### DISCUSSION THREAD

**Discussion by hussainkadori** [FreeOpcUa/python-opcua#1401] [Q&A] [answered]
Hi,
I tried to subscribe to a variable in PLC has 40ms data change interval for plotting, is this possible?

**Comment by AndreasHeine** [comment, marked as answer]
it depends mostly on the plc (min supported sample interval) and the rest of the python code (40 ms is tough)

for questions feel free to use the discoussions feature! -> https://github.com/FreeOpcUa/python-opcua/discussions

**Comment by zerox1212** [comment]
I doubt you will get 40ms refresh on a heavy protocol like UA running on a slow platform like python.

You would likely have to change your strategy, such as subscribing/reading an array containing the last few seconds.

**Comment by oroulet** [comment]
Depends on plc and load on client. But subscription have in fact very little overhead over sendinh bytes on a socket, so in theory it should work.
But my experience with bechhoff plc for example is that they never deliver more than 30hz
